src/olympia/versions/tests/test_models.py

```diff
@@ -199,6 +199,159 @@ def test_latest_public_compatible_with(self):
         ]
         assert list(qs) == expected_versions
 
+    def test_latest_public_compatible_with(self):
+        addon = addon_factory(
+            version_kw={'min_app_version': '57.0', 'max_app_version': '57.*'},
+        )
+        appversions = {
+            'min': version_int('58.0'),
+            'max': version_int('58.0'),
+        }
+        qs = Version.objects.latest_public_compatible_with(amo.FIREFOX.id, appversions)
+        # We should get 4 joins:
+        # - applications_versions
+        # - appversions (min)
+        # - appversions (max)
+        # - files (status and strict_compatibility)
+        assert str(qs.query).count('JOIN') == 4
```

Comment:
Yeah the alternative is to use private APIs (`qs.query.table_map` or `qs.query.alias_map`, but both require you to do manipulate the result to find the join count) to do this... It would be more precise - we could extract the exact table names and even their aliases - but I'm not sure it's worth it.